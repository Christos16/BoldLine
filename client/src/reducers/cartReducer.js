import {
  TOGGLE_CART_HIDDEN,
  ADD_ITEM,
  CLEAR_ITEM_FROM_CART,
  REMOVE_ITEM,
  ADD_QUANTITY,
  DECREASE_QUANTITY
} from '../../src/action/types';

import { addItemToCart, removeItemFromCart } from '../action/cartUtils';

const initialState = {
  hidden: true,
  cartItems: []
};

const cartReducer = (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_CART_HIDDEN:
      return {
        ...state,
        hidden: !state.hidden
      };
    case ADD_ITEM:
      return {
        ...state,
        cartItems: addItemToCart(state.cartItems, action.payload)
      };
    case CLEAR_ITEM_FROM_CART:
      return {
        ...state,
        cartItems: state.cartItems.filter(
          cartItem => cartItem.id !== action.payload
        )
      };
    case REMOVE_ITEM:
      return {
        ...state,
        cartItems: removeItemFromCart(state.cartItems, action.payload)
      };

    default:
      return state;
  }
};

export default cartReducer;
