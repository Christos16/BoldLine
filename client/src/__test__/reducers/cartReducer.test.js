import cartReducer from '../../reducers/cartReducer';
import {
  ADD_ITEM,
  CLEAR_ITEM_FROM_CART,
  REMOVE_ITEM, // Only ones related to the reducer being tested
  GET_PRODUCTS
} from '../../action/types';

//Setting the initial state to 0

describe('select_cardReducer', () => {
  describe('INITIAL_STATE', () => {
    test('is correct', () => {
      const action = { type: 'dummy_action' };
      const initialState = { hidden: true, cartItems: [] };

      expect(cartReducer(undefined, action)).toEqual(initialState);
    });
  });
  /*
  describe('ADD_ITEM', () => {
    it('Should add item into cart', () => {
      const products = [
        {
          size: 1,
          price: 25,
          menCategory: 1,
          womanCategory: 1,
          _id: '12345',
          imagePath: 'http://static.zara',
          title: 'Grey Pullover',
          description: 'Grey Pullover with pockets',
          color: 'Stone',
          quantity: 2,
          department: 'Men'
        }
      ];

      const newState = cartReducer(undefined, {
        type: GET_PRODUCTS,
        payload: products
      })
      expect(newState).toEqual(products)
    });

    test('Should decrease the selected product by id from Cart', () => {
      const product = [
        {
          size: 1,
          price: 25,
          menCategory: 1,
          womanCategory: 1,
          _id: '12345',
          imagePath: 'http://static.zara',
          title: 'Grey Pullover',
          description: 'Grey Pullover with pockets',
          color: 'Stone',
          quantity: 1,
          department: 'Men'
        },
        {
          size: 1,
          price: 25,
          menCategory: 1,
          womanCategory: 1,
          _id: '12345',
          imagePath: 'http://static.zara',
          title: 'Grey Pullover',
          description: 'Grey Pullover with pockets',
          color: 'Stone',
          quantity: 0,
          department: 'Men'
        }
      ];
      const action = { type: REMOVE_ITEM, payload: [product] };
      expect(cartReducer(undefined, action)).toMatchSnapshot();
    });

    test('Should remove the selected product by id from Cart', () => {
      const product = {};
      const action = { type: CLEAR_ITEM_FROM_CART, payload: [product] };
      expect(cartReducer(undefined, action)).toMatchSnapshot();
    });
  });

 
  describe('GET_PRODUCTS', () => {
    test('returns the correct state', () => {
      const action = {
        type: GET_PRODUCTS,
        payload: [
          {
            size: 1,
            price: 25,
            menCategory: 1,
            womanCategory: 1,
            _id: '12345',
            imagePath: 'http://static.zara',
            title: 'Grey Pullover',
            description: 'Grey Pullover with pockets',
            color: 'Stone',
            quantity: 0,
            department: 'Men'
          },
          {
            size: 2,
            price: 22,
            menCategory: 2,
            womanCategory: 3,
            _id: '12345',
            imagePath: 'http://stfatic.zara',
            title: 'Grey ',
            description: ' pockets',
            color: 'red',
            quantity: 0,
            department: 'Woman'
          }
        ]
      };
      const expectedState = {
        products: [
          {
            size: 1,
            price: 25,
            menCategory: 1,
            womanCategory: 1,
            _id: '12345',
            imagePath: 'http://static.zara',
            title: 'Grey Pullover',
            description: 'Grey Pullover with pockets',
            color: 'Stone',
            quantity: 0,
            department: 'Men'
          },
          {
            size: 2,
            price: 22,
            menCategory: 2,
            womanCategory: 3,
            _id: '12345',
            imagePath: 'http://stfatic.zara',
            title: 'Grey ',
            description: ' pockets',
            color: 'red',
            quantity: 0,
            department: 'Woman'
          }
        ]
      };
      expect(selectReducer(undefined, action)).toEqual(expectedState);
    });
   
  });
  */
});
