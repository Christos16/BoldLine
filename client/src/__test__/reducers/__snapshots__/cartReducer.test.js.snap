// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_cardReducer ADD_ITEM Should add item into cart 1`] = `
Object {
  "cartItems": Array [
    Object {
      "0": Array [
        Object {
          "_id": "12345",
          "color": "Stone",
          "department": "Men",
          "description": "Grey Pullover with pockets",
          "imagePath": "http://static.zara",
          "menCategory": 1,
          "price": 25,
          "quantity": 2,
          "size": 1,
          "title": "Grey Pullover",
          "womanCategory": 1,
        },
      ],
      "quantity": 1,
    },
  ],
  "hidden": true,
}
`;

exports[`select_cardReducer ADD_ITEM Should remove the selected product by id from Cart 1`] = `
Object {
  "cartItems": Array [],
  "hidden": true,
}
`;

exports[`select_reducer ADD_ITEM Should add item into cart 1`] = `
Object {
  "NumberOfProducts": "",
  "product": Object {},
  "products": Array [],
}
`;

exports[`select_reducer ADD_ITEM Should decrease the selected product by id from Cart 1`] = `
Object {
  "NumberOfProducts": "",
  "product": Object {},
  "products": Array [],
}
`;

exports[`select_reducer ADD_ITEM Should remove the selected product by id from Cart 1`] = `
Object {
  "NumberOfProducts": "",
  "product": Object {},
  "products": Array [],
}
`;
